src/main/java/com/aws/greengrass/clientdevices/auth/iot/Certificate.java

```diff
@@ -28,6 +105,16 @@ public String getNamespace() {
 
     @Override
     public Map<String, DeviceAttribute> getDeviceAttributes() {
-        return Collections.singletonMap("CertificateId", new StringLiteralAttribute(getIotCertificateId()));
+        return Collections.singletonMap("CertificateId", new StringLiteralAttribute(getCertificateId()));
+    }
+
+    private static String computeCertificateId(String certificatePem)
+            throws CertificateException, NoSuchAlgorithmException, UnsupportedEncodingException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        X509Certificate cert = (X509Certificate) cf.generateCertificate(
+                new ByteArrayInputStream(certificatePem.getBytes(StandardCharsets.UTF_8)));
```

Comment:
need to close the stream?

Reply:
`ByteArrayInputStream` is GC'd without needing to be closed. But, probably best to do it just in case the stream type changes in the future